app/main/posts/views/filters/filters-dropdown.directive.js

```diff
@@ -40,9 +40,8 @@ function FiltersDropdownController($scope, $state, PostFilters, ModalService, $r
         $scope.dropdownStatus.isopen = !$scope.dropdownStatus.isopen;
     };
     $scope.clearFilters = function () {
-        if (PostFilters.getMode() === 'savedsearch' && PostFilters.getModeId()) {
+        if (_.contains(['savedsearch', 'collection'], PostFilters.getMode()) && PostFilters.getModeId()) {
```

Comment:
Is the point of this check just to change the url?

Reply:
yes... and to do the mode reset to 'all' . It was checking against savedsearch only before that and not resetting the mode when you cleared the filters in a collection.